libs/cdk/src/lib/utils/directives/focusable-item/focusable-item.directive.ts

```diff
@@ -38,64 +43,152 @@ export class DeprecatedFocusableItemDirective extends DeprecatedSelector {}
         return this._focusable;
     }
 
+    /** Whether tabbable child should be focused instead. Default is false. */
+    @Input()
+    set focusChild(val: BooleanInput) {
+        this._focusChild = coerceBooleanProperty(val);
+        this.setTabbable(this._focusable);
+    }
+
+    get focusChild(): boolean {
+        return this._focusChild;
+    }
+
+    /** Function, that returns a string to be announced by screen-reader whenever item which is in grid receives focus. */
+    @Input()
+    cellFocusedEventAnnouncer: (position: FocusableItemPosition) => string = this._defaultItemFocusedEventAnnouncer;
+
+    /** @hidden */
+    _position: FocusableItemPosition;
+
+    /** @hidden */
+    private _focusChild = false;
+
     /** @hidden */
     private _focusable = true;
+
+    /** @hidden */
+    private _tabbableElements = new Map<HTMLElement, number>();
+
+    /** @hidden */
+    private _tabbable = true;
+
     /** @hidden */
-    private readonly _viewInit$ = new BehaviorSubject(false);
+    private _timerId: number | null;
+
+    /** @hidden */
+    @HostBinding('attr.tabindex')
+    get _tabindex(): number {
+        return this._tabbable ? 0 : -1;
+    }
 
     /** @hidden */
     constructor(
         private _elementRef: ElementRef<HTMLElement>,
         private _focusableObserver: FocusableObserver,
-        private _destroy$: DestroyedService
+        private _destroy$: DestroyedService,
+        private _tabbableElementService: TabbableElementService,
+        private _liveAnnouncer: LiveAnnouncer
     ) {
-        super(1);
         this._focusableObserver
-            .observe(this._elementRef)
-            .pipe(
-                tap((isFocusable) => {
-                    if (isFocusable !== this._focusable) {
-                        this.setFocusable(isFocusable);
-                        this._focusable = isFocusable;
-                        this.next(isFocusable);
-                    }
-                }),
-                takeUntil(this._destroy$)
-            )
-            .subscribe();
+            .observe(this._elementRef, false)
+            .pipe(takeUntil(this._destroy$))
+            .subscribe((isFocusable) => {
+                if (isFocusable !== this.fdkFocusableItem) {
+                    this.fdkFocusableItem = isFocusable;
+                }
+            });
+    }
+
+    /** Element reference. */
+    elementRef(): ElementRef<HTMLElement> {
+        return this._elementRef;
+    }
+
+    /** Set tabbable state */
+    setTabbable(state: boolean): void {
+        this._tabbable = state;
+
+        if (!this.focusChild) {
+            return;
+        }
+
+        if (state) {
+            this._enableTabbableElements();
+        } else {
+            this._disableTabbableElements();
+        }
     }
 
     /** @hidden */
-    ngAfterViewInit(): void {
-        this._viewInit$.next(true);
+    @HostListener('focusin')
+    async _onFocusin(): Promise<void> {
+        if (this._timerId != null) {
```

Comment:
What is the purpose of `_timerId`? I'm sure there's a reason I just don't know. Change detection related?

Reply:
That's for supporting multiple focusable elements inside a cell, when we come to the cell we should focus first focusable child and read a11y text but then when tabbing between focusable elements inside a single cell we should not manipulate focus neither read a11y text again.